File: src/models/monitor.ts
export interface MonitorInfo {
  systemMonitorId: string;
  name: string;
  width: number;
  height: number;
  positionX: number;
  positionY: number;
  primary: boolean;
}

export interface AppStatus {
  appDataDirectory: string;
  databasePath: string;
  platform: string;
  schemaVersion: number;
}

export interface MonitorSlice {
  systemMonitorId: string;
  canvasX: number;
  canvasY: number;
  width: number;
  height: number;
}

export interface MonitorLayout {
  layoutHash: string;
  originX: number;
  originY: number;
  width: number;
  height: number;
  slices: MonitorSlice[];
}

File: src-tauri/src/models/monitor.rs
use serde::Serialize;

/// Platform-independent snapshot of one currently attached display.
#[derive(Clone, Debug, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct MonitorInfo {
    pub system_monitor_id: String,
    pub name: String,
    pub width: u32,
    pub height: u32,
    pub position_x: i32,
    pub position_y: i32,
    pub primary: bool,
}
